eole/decoders/transformer_lm_decoder.py

```diff
@@ -91,6 +102,9 @@ def __init__(
     ):
         super(TransformerLMDecoder, self).__init__(model_config)
 
+        if model_config.position_encoding_type == PositionEncodingType.Rotary:
+            self.rope = RotaryPosition(model_config)
+
```

Comment:
what about setting self.rope to none otherwise, then we avoid the if test below, line 130

Reply:
Hmm we still would have to test if None to set position_embeddings to None.
An alternative may be to have some small method to encapsulate this condition, but we would still need some form of condition.

Reply:
my mistake you're right. in fact another way to do it would be to define a lambda function in the init to avoid testing and use a single testing in init. something like: https://github.com/eole-nlp/eole/blob/main/eole/modules/transformer_mlp.py#L58-L62

Reply:
That's a tiny bit more efficient indeed, but a bit less readable. I think we can keep it as is, and revisit at some point, e.g. if new position encoding methods arise...